internal/lifecycle/readme_test.go

```diff
@@ -0,0 +1,504 @@
+	//		},
+	//	},
+	//	cmds: []*exec.Cmd{
+	//		exec.Command("gcloud", "--quiet", "run", "services", "deploy", "unique_service_name", "--image", "gcr.io/unique/tag"),
+	//	},
+	//	serviceName: "unique_service_name",
+	//	gcrURL: "gcr.io/unique/tag",
+	//},
+
+	// replace Cloud Run service name and GCR URL with provided inputs and expand environment variables
+	{
+		inStr: "[//]: # ({sst-run-unix})\n" +
+			"```\n" +
+			"gcloud run services deploy hello_world --image=gcr.io/hello/world --add-cloudsql-instances=${TEST_CLOUD_SQL_CONNECTION}\n" +
+			"```\n",
+		codeBlocks: []codeBlock{
+			[]string{
+				"gcloud run services deploy hello_world --image=gcr.io/hello/world --add-cloudsql-instances=${TEST_CLOUD_SQL_CONNECTION}",
+			},
+		},
+		cmds: []*exec.Cmd{
+			exec.Command("gcloud", "--quiet", "run", "services", "deploy", "unique_service_name", "--image=gcr.io/unique/tag", "--add-cloudsql-instances=project:region:instance"),
+		},
+		env: map[string]string{
+			"TEST_CLOUD_SQL_CONNECTION": "project:region:instance",
+		},
+		serviceName: "unique_service_name",
+		gcrURL:      "gcr.io/unique/tag",
+	},
+
+	// replace Cloud Run service name provided name in command with multiline arguments
+	{
+		inStr: "[//]: # ({sst-run-unix})\n" +
+			"```\n" +
+			"gcloud run services update hello_world --add-cloudsql-instances=\\\n" +
+			"project:region:instance\n" +
+			"```\n",
+		codeBlocks: []codeBlock{
+			[]string{
+				"gcloud run services update hello_world --add-cloudsql-instances=\\",
+				"project:region:instance",
+			},
+		},
+		cmds: []*exec.Cmd{
+			exec.Command("gcloud", "--quiet", "run", "services", "update", "unique_service_name", "--add-cloudsql-instances=project:region:instance"),
+		},
+		serviceName: "unique_service_name",
+		gcrURL:      "gcr.io/unique/tag",
+	},
+
+	// replace Cloud Run service name provided name and expand environment variables in command with multiline arguments
+	{
+		inStr: "[//]: # ({sst-run-unix})\n" +
+			"```\n" +
+			"gcloud run services update hello_world --add-cloudsql-instances=\\\n" +
+			"${TEST_CLOUD_SQL_CONNECTION}\n" +
+			"```\n",
+		codeBlocks: []codeBlock{
+			[]string{
+				"gcloud run services update hello_world --add-cloudsql-instances=\\",
+				"${TEST_CLOUD_SQL_CONNECTION}",
+			},
+		},
+		cmds: []*exec.Cmd{
+			exec.Command("gcloud", "--quiet", "run", "services", "update", "unique_service_name", "--add-cloudsql-instances=project:region:instance"),
+		},
+		env: map[string]string{
+			"TEST_CLOUD_SQL_CONNECTION": "project:region:instance",
+		},
+		serviceName: "unique_service_name",
+		gcrURL:      "gcr.io/unique/tag",
+	},
+
+	// markdown file input to codeBlocks test
+	{
+		inFileName: "readme_test.md",
+		cmds: []*exec.Cmd{
+			exec.Command("echo", "hello", "world"),
+			exec.Command("echo", "line", "one"),
+			exec.Command("echo", "line", "two"),
+		},
+	},
+}
+
+func TestToCommands(t *testing.T) {
+	for i, tc := range tests {
```

Comment:
Rather than iterate through each list of tests for each section, you should probably just have a different slice for each test.